packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx

```diff
@@ -59,6 +67,47 @@ const ReleaseDetailsPage = () => {
     setIsPopoverVisible((prev) => !prev);
   };
 
+  const toggleEditReleaseDialog = () => {
+    setEditReleaseDialogIsShown((prev) => !prev);
+  };
+
+  const openAddEditReleaseDialog = () => {
+    toggleEditReleaseDialog();
+    handleTogglePopover();
+  };
+
+  const [updateRelease, { isLoading }] = useUpdateReleaseMutation();
+
+  const handleEditRelease = async (values: FormValues) => {
+    const response = await updateRelease({
+      id: releaseId,
+      name: values.name,
+    });
+    if ('data' in response) {
+      // When the response returns an object with 'data', handle success
+      toggleNotification({
+        type: 'success',
+        message: formatMessage({
+          id: 'content-releases.modal.release-updated-notification-success',
+          defaultMessage: 'Release updated.',
+        }),
+      });
+    } else if (isAxiosError(response.error)) {
+      // When the response returns an object with 'error', handle axios error
+      toggleNotification({
+        type: 'warning',
+        message: formatAPIError(response.error),
+      });
+    } else {
+      // Otherwise, the response returns an object with 'error', handle a generic error
+      toggleNotification({
+        type: 'warning',
+        message: formatMessage({ id: 'notification.error', defaultMessage: 'An error occurred' }),
```

Comment:
We starting to have this two conditionals a lot in all the plugin, maybe worth to move this to a util function that receives the error as an utils?

Something like:
```
// /utils/handleReleaseError.ts

export const handleReleaseError = (error) => {
  if (isAxiosError(error) { return toggle.... }
  
  return toggle...
}
```

Reply:
I'd be mindful of this, what happens if you have to handle different errors e.g. validation from the server? you also need to pass `toggleNotification` around because it doesn't exist outside of react

Reply:
Thanks Josh, I tried yesterday to find a solution to abstract this part but tbh I didn t find one and also as you said we can have different errors for different APIs, I would stay with the actual code

Reply:
I think that's a great decision :)